pkg/autohealing/cloudprovider/cloudprovider.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudprovider
+
+import (
+	log "k8s.io/klog"
+
+	"k8s.io/cloud-provider-openstack/pkg/autohealing/config"
+	"k8s.io/cloud-provider-openstack/pkg/autohealing/healthcheck"
+)
+
+var (
+	providers = make(map[string]RegisterFunc)
+)
+
+// CloudProvider is an abstract, pluggable interface for cloud providers.
+type CloudProvider interface {
```

Comment:
does this intended support other cloud providers than OpenStack? should this be just in the Kubernetes org?

Reply:
We could move to kubernetes.org after there are other providers implemented and maintainers from other clouds, but for now, there is no plan yet. I'm very happy that people could help to implement other cloud providers and we could find a better place for the service then.